src/main/java/MorseCodeConverter.java

```diff
@@ -1,4 +1,61 @@
 public class MorseCodeConverter
 {
+    private final MorseCode morseCode;
 
+    public MorseCodeConverter()
+    {
+        this.morseCode = new MorseCode();
+    }
+
+    public String toMorseCode(String englishExpression)
+    {
+        englishExpression = englishExpression.toUpperCase();
+        String morseExpression = "";
+        String[] englishWords = englishExpression.split(" ");
+        for (String word : englishWords)
+        {
+            String[] englishLetters = word.split("");
+
+            for (String letter : englishLetters)
+            {
+                for (Character key : morseCode.getMorseCharacters().keySet())
+                {
+                    String keyString = key.toString();
+                    if (letter.equals(keyString))
+                    {
+                        morseExpression += morseCode.getMorseCharacters().get(key) + " ";
+                        break;
+                    }
+                }
```

Comment:
You aren't using the hashmap correctly

Reply:
In what way am I doing this incorrectly? Here only or also in the toEnglish method?

Reply:
You shouldn't need to loop through the hashmap.

You should use the get method

Reply:
ohhhh right!